### ISSUE TITLE
🐛 Issue with 'ignore' functionality in version 1.5

### ISSUE THREAD

**Issue by riya-amemiya** [biomejs/biome#1580]
### Environment information

```block
CLI:
  Version:                      1.5.2
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.6.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "bun/1.0.23"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

In version 1.4.1, the 'ignore' functionality worked as expected.
However, starting from version 1.5, the 'ignore' option no longer functions correctly.
While it doesn't produce warnings, it still processes files and directories included in the 'ignore' list, which leads to a decrease in processing speed.

### Expected result

The 'ignore' option should exclude the specified directories and files from being processed, just as it did in version 1.4.1.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by riya-amemiya** [comment]
For example, the following is an example of the behavior when the 'module' directory is set in the 'ignore' configuration but is not being ignored:

```shell
❯ bun run biome check .
Checked 800 file(s) in 33ms

❯ bun run biome check . // Removed only the 'module' directory without changing the biome configuration
Checked 276 file(s) in 23ms
```

In this scenario, even though the 'module' directory is specified in the 'ignore' list, it is not excluded from processing, as demonstrated by the number of files checked in both cases.

**Comment by ematipico** [comment]
Hi @riya-amemiya 

You gave us zero information to work with, so it's literally impossible for us to help you to triage the issue.

Please, share with us a minimal reproduction. You can use our codesandbox template too, you can find the link in the issue form

**Comment by riya-amemiya** [comment]
@ematipico 

I apologize for any inconvenience, and thank you for your polite response. I will share the minimum configuration.

https://github.com/riya-amemiya/biome-sandbox

**Comment by Conaclos** [comment]
Hi @riya-amemiya.

I checked your reproduction.
Assuming that you mean `src` instead of `module` in your reproduction, Biome linter/formatter seems to correctly ignore `src/index.ts`.
However, the reported number of processed files is indeed incorrect.

The bug is not present if the files/directories are ignored globally using `files.ingnore`.

**Comment by amoreaulemay** [comment]
I had a similar issue upgrading to 1.5.2 (from 1.4.1). I have this section in my biome.js:

```json
"files": {
    "ignoreUnknown": true,
    "ignore": ["node_modules/*", "./node_modules/*"]
  }
```
And biome still included node_modules and gave a bunch of symlink errors.

**Comment by amoreaulemay** [comment]
> I had a similar issue upgrading to 1.5.2 (from 1.4.1). I have this section in my biome.js:
> 
> ```json
> "files": {
>     "ignoreUnknown": true,
>     "ignore": ["node_modules/*", "./node_modules/*"]
>   }
> ```
> 
> And biome still included node_modules and gave a bunch of symlink errors.

I actually found a fix for my issue:

For biome 1.4.1 I had this pnpm script in my package.json:

`biome check --error-on-warnings --formatter-enabled=true --linter-enabled=true --organize-imports-enabled=true ./**/*`

And for 1.5.2 I had to modify it to this:

`biome check --error-on-warnings --formatter-enabled=true --linter-enabled=true --organize-imports-enabled=true .`

So the problem was the wildcard at the end.

**Comment by ematipico** [comment]
Have you tried `node_modules/**` instead?

**Comment by amoreaulemay** [comment]
I just did, that doesn't fix it. Removing the `./**/*` for `.` was the real fix for me. Then it respects my ignore.

**Comment by Conaclos** [comment]
> I just did, that doesn't fix it. Removing the `./**/*` for `.` was the real fix for me. Then it respects my ignore.

`./**/*` is expanded by your shell to all files. Thus, this passed to Biome all files, including the one in `node_modules` forcing Biome to take a look at these paths.
